src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6474,6 +6474,13 @@ private BoundExpression BindLeftIdentifierOfPotentialColorColorMemberAccess(Iden
                                 Debug.Assert(!leftType.IsDynamic());
                                 Debug.Assert(IsPotentialColorColorReceiver(left, leftType));
 
+                                if (leftSymbol is SourceFieldSymbolWithSyntaxReference fieldLeft)
+                                {
+                                    boundValue = RebindColorColorConstField(left, boundValue, fieldLeft);
+                                    leftSymbol = boundValue.ExpressionSymbol;
+                                    ConstantFieldsInProgress.RemoveDepencency(fieldLeft);
```

Comment:
>ConstantFieldsInProgress.RemoveDepencency(fieldLeft); [](http://example.com/codeflow?start=36&length=53)

This looks fragile to me. How do we know that ```BindIdentifier``` above was the only reason for the field to be in the set? What if after "rebinding" the reference should still remain in the set?

Reply:
This is deeper than I thought; the more the hours were passing, the more convinced I am that we need to approve a spec change, which will make total sense for this line of code.

Essentially, on top of the field initialization, we could entirely exclude the field itself that is being initialized, in order to properly bind to another symbol like a namespace, or type. I will open up a discussion/issue for this matter following further contact.

Reply:
> Essentially, on top of the field initialization, we could entirely exclude the field itself that is being initialized, in order to properly bind to another symbol like a namespace, or type. I will open up a discussion/issue for this matter following further contact.

I think changing the spec is premature. Even if this particular attempt to address the issue isn't something we are comfortable with, it doesn't mean there is no alternative way to approach the problem. We can "sit" on the issue a little longer and let someone from the compiler team to take a closer look once we have time.